docs: add AGENTS.md with Cursor Cloud development instructions (#35)
seqeralabs/custom-studios-examples: docs/agents/custom-studios.md

### Option A — Extend a built-in template (Conda + Wave)

Use when you only need extra packages on an existing template (Jupyter, R IDE, VS Code, Xpra):

1. In Seqera Platform: **Studios → Add Studio**.
2. Choose a base template and supply a **Conda environment specification** on the second screen.
3. Wave builds a new image with your dependencies layered on the template.

Example from the blog: add IGV to the Xpra template via Conda.

**No custom Dockerfile required.** Best for small dependency additions.

### Option B — Custom Dockerfile (this repo's model)

Use when the app is not covered by templates, or you need full control over the runtime (Python/R base, arbitrary tools, custom startup logic).

Every example in this repo follows Option B.

---

## Standard Dockerfile structure

All studios in this repo use the same multi-stage pattern. Required steps:

### 1. Pull connect-client (multi-stage)

Connect mediates communication between the app and Seqera Platform and includes Fusion (POSIX-like cloud storage access).

```dockerfile
ARG CONNECT_CLIENT_VERSION=0.9
FROM public.cr.seqera.io/platform/connect-client:${CONNECT_CLIENT_VERSION} AS connect
```

> Pass the version at build time: `--build-arg CONNECT_CLIENT_VERSION=0.9`. Dockerfiles in this repo default to `0.9`.

Reviewer: Defaults aren't uniformly `0.9`. The `master` example dirs pin `0.9`, but the `marimo`, `streamlit`, `ttyd`, and `shinyngs` `.seqera/` branches pin `0.12`. Since the same studio can deploy with a different version depending on path, better to point readers at the Dockerfile rather than assert one value.

```suggestion
Pass the version at build time, e.g. `--build-arg CONNECT_CLIENT_VERSION=0.9`. Defaults vary across the repo: the `master` example directories pin `0.9`, while several `.seqera/` branches (`marimo`, `streamlit`, `ttyd`, `shinyngs`) pin `0.12`. Check the `ARG CONNECT_CLIENT_VERSION` default in the specific Dockerfile rather than assuming a single value.
```

### 2. Choose an application base image

Install your app on a base that supports package management (`apt-get` or `yum` required for `connect-client --install`):

| Pattern | Example in this repo |
|---------|----------------------|
| Python slim + pip/uv | `marimo/`, `streamlit/`, `cellxgene/` |
| Ubuntu + micromamba | `shiny-simple-example/` |
| Existing community image + inject tools | `ttyd/` (SAMtools base + TTYD) |

Stripped images without `apt-get`/`yum` cannot run `connect-client --install`.

### 3. Define environment variables (optional)

Expose tunable defaults with `ENV`. Users override them when launching the Studio in Platform:

```dockerfile
ENV DATASET_FILE=s3://cellxgene_datasets/pbmc3k.h5ad
ENV DATASET_TITLE="PBMCs 3k test dataset"
```

See [Environment variables](#environment-variables) below for per-studio variables in this repo.

### 4. Install connect-client in the final stage

```dockerfile
ARG CONNECT_CLIENT_VERSION
LABEL io.seqera.connect.version="${CONNECT_CLIENT_VERSION}"
COPY --from=connect /usr/bin/connect-client /usr/bin/connect-client
RUN /usr/bin/connect-client --install
```

> Some Python-based images also install `btrfs-progs` from Debian backports (see `marimo/Dockerfile`, `cellxgene/Dockerfile`) — follow neighboring examples when using `python:*-slim`.

Reviewer: `streamlit/Dockerfile` also installs `btrfs-progs` from backports — all three Python-slim images do.

```suggestion
Some Python-based images also install `btrfs-progs` from Debian backports (see `marimo/Dockerfile`, `cellxgene/Dockerfile`, `streamlit/Dockerfile`) — follow neighboring examples when using `python:*-slim`.
```

  bucket_name <- strsplit(cloud_path, '/')[[1]][1]
  object_path <- sub(paste0('^', bucket_name, '/'), '', cloud_path)
  file_path <- paste0('/workspace/data/', bucket_name, '/', object_path)
} else {
  file_path <- data_path
}
```

Choose Bash (startup script) or app-level translation depending on how your tool accepts input paths.

### Streamlit + Fusion

`streamlit/` clones [MultiQC/example-streamlit](https://github.com/MultiQC/example-streamlit). Users load data via **Server Paths** pointing at `/workspace/data/<link>/file.zip` after mounting — no app code changes needed for Fusion access.

---

## Environment variables

Variables defined in the Dockerfile can be overridden at Studio launch in Platform.

| Studio | Variable | Default | Purpose |
|--------|----------|---------|---------|
| CellxGene | `DATASET_FILE` | `s3://cellxgene_datasets/pbmc3k.h5ad` | Input `.h5ad` |
| CellxGene | `DATASET_TITLE` | `PBMCs 3k test dataset` | Display title |
| CellxGene | `USER_DATA_DIR` | `/user-data/cellxgene` | User-generated data |
| CellxGene | `ANNOTATIONS_DIR` | `/user-data/cellxgene` | Annotations |
| Shiny | `DATA_PATH` | `s3://shiny-inputs/data.csv` | Input CSV |

> Marimo, Streamlit, and TTYD use defaults only (no required env vars).

Reviewer: Shinyngs also runs on defaults only — matches the README's env-var note.

```suggestion
Marimo, Streamlit, Shinyngs, and TTYD use defaults only (no required env vars).
```

---

## Build and validate locally

```bash
cd <studio-directory>
sudo docker build --platform=linux/amd64 \
  --build-arg CONNECT_CLIENT_VERSION=0.9 \
  -t my-studio .
```

Override the entrypoint for local runs (connect-client needs Platform). Each studio README documents the local command.

CI (`.github/workflows/docker-pr.yml`) builds changed Dockerfiles on PR and runs Trivy scans.

---

## Deploy to Seqera Platform

### Pre-built container image

1. Push the image to a registry accessible from Platform (GHCR, ECR, etc.).
2. **Studios → Add Studio → Prebuilt container image**.
3. Enter the image URI (e.g. `ghcr.io/seqeralabs/custom-studios-examples/shiny:latest`).
4. **Compute and Data**: select compute env, CPU/memory/GPU, **Mount data** for required buckets.
5. **General config**: set name, description, and any env vars (`DATA_PATH`, `DATASET_FILE`, …).
6. Review and **Add → Start**.

Pre-built images for this repo:

```
ghcr.io/seqeralabs/custom-studios-examples/marimo:latest
ghcr.io/seqeralabs/custom-studios-examples/cellxgene:latest
ghcr.io/seqeralabs/custom-studios-examples/streamlit:latest
ghcr.io/seqeralabs/custom-studios-examples/shiny:latest

> ghcr.io/seqeralabs/custom-studios-examples/ttyd:latest

Reviewer: The README lists `shinyngs:latest` in its pre-built image block; this list omits it. Suggestion below matches the README.

> [!NOTE]
> `shinyngs` isn't in `docker-release.yml`'s release choices, so this image may not actually be published yet — reconcile the README/workflow rather than trusting either blindly.

```suggestion
ghcr.io/seqeralabs/custom-studios-examples/shinyngs:latest
ghcr.io/seqeralabs/custom-studios-examples/ttyd:latest
```
